### ISSUE TITLE
Combat screen, logs list, wrong way to indicate spell damage

### ISSUE THREAD

**Issue by LeHerosInconnu** [ihhub/fheroes2#2797]
In the original game, damage is indicated according to the hero's Spell Power.
This is not the case in fheroes2.

Example in the original game:
The Armageddon spell does 50 damage by Spell Power.
The hero has 16 Spell Power.
What is indicated is the damage caused according to the hero's Spell Power:
"The armageddon does 800 damage."

![Damage display H2 01](https://user-images.githubusercontent.com/43583994/107817511-8cc19f00-6d76-11eb-95f4-49da7e787618.png)

![Damage display H2 02](https://user-images.githubusercontent.com/43583994/107817530-90edbc80-6d76-11eb-928f-1c9a9a0750b9.png)

Example in fheroes2:
The Armageddon spell does 50 damage by Spell Power.
The hero has 2 Spell Power.
What is indicated is the damage caused to all the troops:
"The Armageddon does 600 damage."

![Damage display fh2 01](https://user-images.githubusercontent.com/43583994/107817585-a06d0580-6d76-11eb-84c3-8b4a77414c51.png)

![Damage display fh2 02](https://user-images.githubusercontent.com/43583994/107817601-a4992300-6d76-11eb-9e4f-c636eefd69d7.png)

This way of presenting damage misleads the player; it lets the player think that the spell does a lot of damage when this is not really the case.

For all spells, the damage should be indicated according to the hero's Spell Power, as in the original game.

**Comment by Branikolog** [comment]
I agree, that this is not the same implementation, comparing to original game. 
But in fheroes2 we can add spell damage in spell book info, so player understands, how much damage each troop would take. 
I believe, the total damage is more useful information, as player would know, if a spell was effective .

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> I agree, that this is not the same implementation, comparing to original game.
> But in fheroes2 we can add spell damage in spell book info, so player understands, how much damage each troop would take.
> I believe, the total damage is more useful information, as player would know, if a spell was effective enough.

I suggested your proposal in this thread:

> @Branikolog.
> 
> I think that, as you suggested before, the best solution would be to specify the total damage at the same time:
> 
> "The Death Ripple spell does 30 damage to 5 troops of living creatures (total: 150)."
> 
> Or:

> "The Death Ripple spell does 30 damage to 5 troops of living creatures for a total of 150 damage."

_Originally posted by @LeHerosInconnu in https://github.com/ihhub/fheroes2/pull/2730#discussion_r575693185_

**Comment by Branikolog** [comment]
@LeHerosInconnu 
I'm really sorry for being too opinionated.  Your issue above is absolutely right and relevant. I think we should make the original machanics for start and only then discuss some improvements.
Also I have a question for you regarding the issue above.
You were talking about mass spells. But what about area spells (ex. Ice Ring). Are they working the same currently and require fix for log description of the damage dealt? Does a log description of this area spells also show correct values if a hero has artifacts that affect damage of the spell?

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> @LeHerosInconnu
> I'm really sorry for being too opinionated. Your issue above is absolutely right and relevant. I think we should make the original machanics for start and only then discuss some improvements.

No worries. :)

> Also I have a question for you regarding the issue above.
> You were talking about mass spells. But what about area spells (ex. Ice Ring). Are they working the same currently and require fix for log description of the damage dealt? Does a log description of this area spells also show correct values if a hero has artifacts that affect damage of the spell?

I will check this later, and I will precise it if necessary.

**Comment by LeHerosInconnu** [comment]
> > Also I have a question for you regarding the issue above.
> > You were talking about mass spells. But what about area spells (ex. Ice Ring). Are they working the same currently and require fix for log description of the damage dealt? Does a log description of this area spells also show correct values if a hero has artifacts that affect damage of the spell?
> 
> I will check this later, and I will precise it if necessary.

In the original game, spell protection artifacts are correctly taken into account for the display of damage.
In fheroes2, this is also the case.

In original game:

The hero has 2 Spell Power:
![vlcsnap-2021-03-06-11h46m06s878](https://user-images.githubusercontent.com/43583994/110205385-ec8fff00-7e77-11eb-9301-d60ac01e880a.png)

Damage with Lightning Helm:
![vlcsnap-2021-03-06-11h46m21s432](https://user-images.githubusercontent.com/43583994/110205410-0c272780-7e78-11eb-825a-c932b9118b2a.png)

Damage with Lighting Rod and Lightning Helm:
![vlcsnap-2021-03-06-11h47m31s472](https://user-images.githubusercontent.com/43583994/110205414-16492600-7e78-11eb-81e6-7da30f1ab4c9.png)

Damage with Fire Cloak:
![vlcsnap-2021-03-06-11h51m44s196](https://user-images.githubusercontent.com/43583994/110205420-2103bb00-7e78-11eb-9d49-8724c2ba084d.png)

In fheroes2:

Damage with Lighting Rod and Lightning Helm:
![vlcsnap-2021-03-06-11h59m27s491](https://user-images.githubusercontent.com/43583994/110205433-2a8d2300-7e78-11eb-9739-b48285a878b3.png)

Damage on 1 troop with Fire Cloak:
![vlcsnap-2021-03-06-12h00m34s577](https://user-images.githubusercontent.com/43583994/110205453-3e388980-7e78-11eb-94df-b033e7f0603b.png)

Damage on 2 troops with Fire Cloak:
![vlcsnap-2021-03-06-12h01m12s327](https://user-images.githubusercontent.com/43583994/110205460-47c1f180-7e78-11eb-8bdd-767dd53fac64.png)

In fheroes2, it would also be nice to specify (add a sentence) when artifacts modify the damage given (Lighting Rod, etc.) and received (Lightning Helm, etc.).
I'll add it to the discussion on description logs for all damage spells: https://github.com/ihhub/fheroes2/discussions/2873.

**Comment by Branikolog** [comment]
@LeHerosInconnu 
Thanks, for your precision!
What about the cases, when some creatures have artifact affected them and some are not? So both of creatures are hurt by the same spell.

**Comment by LeHerosInconnu** [comment]
@Branikolog.

> @LeHerosInconnu
> Thanks, for your precision!
> What about the cases, when some creatures have artifact affected them and some are not? So both of creatures are hurt by the same spell.

I think this should be specified with several sentences. :)
